File: src/database/seeders/TaxStatusSeeder.php
<?php

namespace Database\Seeders;

use Illuminate\Database\Seeder;
use App\Models\TaxStatus;

class TaxStatusSeeder extends Seeder
{
    /**
     * Run the database seeds.
     */
    public function run(): void
    {
        $taxStatuses = [
            ['code' => '1', 'name' => 'IVA Responsable Inscripto', 'is_active' => true],
            ['code' => '2', 'name' => 'IVA Responsable no Inscripto', 'is_active' => true],
            ['code' => '3', 'name' => 'IVA no Responsable', 'is_active' => true],
            ['code' => '4', 'name' => 'IVA Sujeto Exento', 'is_active' => true],
            ['code' => '5', 'name' => 'Consumidor Final', 'is_active' => true],
            ['code' => '6', 'name' => 'Responsable Monotributo', 'is_active' => true],
            ['code' => '7', 'name' => 'Sujeto no Categorizado', 'is_active' => true],
            ['code' => '8', 'name' => 'Proveedor del Exterior', 'is_active' => true],
            ['code' => '9', 'name' => 'Cliente del Exterior', 'is_active' => true],
            ['code' => '10', 'name' => 'IVA Liberado – Ley Nº 19.640', 'is_active' => true],
            ['code' => '11', 'name' => 'IVA Responsable Inscripto – Agente de Percepción', 'is_active' => true],
            ['code' => '12', 'name' => 'Pequeño Contribuyente Eventual', 'is_active' => true],
            ['code' => '13', 'name' => 'Monotributista Social', 'is_active' => true],
            ['code' => '14', 'name' => 'Pequeño Contribuyente Eventual Social', 'is_active' => true],
        ];

        foreach ($taxStatuses as $taxStatus) {
            TaxStatus::firstOrCreate(
                ['code' => $taxStatus['code']],
                $taxStatus
            );
        }
    }
}

File: backend-dotnet/src/Ikhana.Infrastructure/Data/Seeders/TaxStatusSeeder.cs
using Ikhana.Application.Common.Interfaces;
using Ikhana.Domain.Entities;
using Ikhana.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace Ikhana.Infrastructure.Data.Seeders;

public class TaxStatusSeeder : IDataSeeder
{
    private readonly AppDbContext _context;

    public int Order => 3;

    public TaxStatusSeeder(AppDbContext context)
    {
        _context = context;
    }

    public async Task SeedAsync(CancellationToken cancellationToken = default)
    {
        if (await _context.Set<TaxStatus>().AnyAsync(cancellationToken))
            return;

        var statuses = new (string Code, string Name)[]
        {
            ("1", "Responsable Inscripto"),
            ("2", "Responsable no Inscripto"),
            ("3", "No Responsable"),
            ("4", "Sujeto Exento"),
            ("5", "Consumidor Final"),
            ("6", "Responsable Monotributo"),
            ("7", "Sujeto no Categorizado"),
            ("8", "Proveedor del Exterior"),
            ("9", "Cliente del Exterior"),
            ("10", "Liberado - Ley N° 19.640"),
            ("11", "Agente de Percepción"),
            ("12", "Pequeño Contribuyente Eventual"),
            ("13", "Monotributista Social"),
            ("14", "Pequeño Contribuyente Eventual Social"),
        };

        foreach (var (code, name) in statuses)
        {
            _context.Set<TaxStatus>().Add(new TaxStatus
            {
                Code = code,
                Name = name,
                IsActive = true,
            });
        }

        await _context.SaveChangesAsync(cancellationToken);
    }
}
